src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -104,6 +104,11 @@
     @JsonView(QuizView.Before.class)
     private Course course;
 
+    @OneToMany(mappedBy = "exercise", fetch = FetchType.EAGER, cascade = CascadeType.REMOVE)
```

Comment:
please do not use FetchType.EAGER

Reply:
@OneToMany(mappedBy = "exercise", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderColumn
    @JsonIgnoreProperties(value = "exercise", allowSetters = true)
    @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
    private List< GradingInstruction > structuredGradingInstructions = new ArrayList<>();